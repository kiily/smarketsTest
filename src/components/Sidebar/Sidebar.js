import React from 'react';

const sidebar = (props) => {
    
    return (
        <aside>
            
        </aside>
    )
}