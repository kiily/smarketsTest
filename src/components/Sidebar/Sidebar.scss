@import '../../styles/variables';

// Left sidebar
nav.Sidebar-left {
    position: fixed;
    top: 3rem;
    bottom: 0;
    left: 0;
    background-color: $sidebar-grey;
    &.in {
        animation: slidein 0.5s ease-in;
    }
    &.out {
        transform: translateX(-240px);
        animation: slideout 0.5s ease-out;
    }
    .sidebar-container {
        width: 240px;
        .left-menu-quick-links {
            padding: 0rem;
            margin: 5px 0;
            .links-title {
                @include uppercase-text($font-m);
                color: $green;
                padding: 10px 1rem;
                font-weight: 600;
            }
            .menu-item {
                display: flex;
                font-size: $font-l;
                color: $white;
                a {
                    &.active {
                        background: $green-50;
                    }
                    padding: 1rem;
                    width: 100%;
                    font-weight: 500;
                    color: $white;
                    text-decoration: none;
                }
            }
        }   
    }
}

// Right sidebar
aside.Sidebar-right {
    position: fixed;
    top: 3rem;
    right: 0;
    bottom: 0;
    background-color: $sidebar-grey;
    .sidebar-container {
        width: 340px;
        .popular-events {
            .right-nav-header {
                display: flex;
                padding: 1rem 1.25rem;
                text-transform: uppercase;
                font-size: $font-l;
                font-weight: bold;
                height: 3.125rem;
                color: $white;
                span {
                    display: flex;
                    align-items: center;
                }
            }
            .right-nav-content {
                .popular-events-list {
                    margin: 0;
                    padding: 0 10px;
                    list-style-type: none;
                    li > a {
                        display: flex;
                        color: $white;
                        padding: 0.75rem 0.5rem;
                        border-top: 1px solid $border-color;
                        text-decoration: none;
                        &:hover {
                            background-color: $border-color;
                            border-top-color: transparent;
                        }

                        .event-link-content {

                        }
                    }
                }
            }
        }
    }
}

@media only screen and (max-width: 1024px) {
    nav.Sidebar-left {
        z-index: 1;
    }
    aside.Sidebar-right {
        display: none;
    }
}


@keyframes slideout {
    from {
        transform: translate(0);
    }
    to {
        transform: translateX(-240px);
    }
}

@keyframes slidein {
    from {
        transform: translate(-240px);
    }
    to {
        transform: translateX(0);
    }
}