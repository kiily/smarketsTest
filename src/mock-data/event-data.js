export const events = [
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-16T10:09:18.291988Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/championship/2019/04/30/millwall-fc-vs-bristol-city-fc',
    'hidden': false,
    'parent': {
      'name': 'Championship'
    },
    'id': '1925896',
    'inplay_enabled': true,
    'modified': '2019-04-16T10:09:18.291988Z',
    'name': 'Millwall vs. Bristol City',
    'parent_id': '939285',
    'short_name': 'MWL vs. BRC',
    'slug': 'millwall-fc-bristol-city-fc-2019-04-30',
    'special_rules': null,
    'start_date': '2019-04-30',
    'start_datetime': '2019-04-30T18:45:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-17T10:10:36.025237Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/championship/2019/05/01/swansea-vs-derby',
    'hidden': false,
    'parent': {
      'name': 'Championship'
    },
    'id': '1934371',
    'inplay_enabled': true,
    'modified': '2019-04-17T10:10:36.025237Z',
    'name': 'Swansea City vs. Derby County',
    'parent_id': '939285',
    'short_name': 'SWA vs. DER',
    'slug': 'swansea-derby-2019-05-01',
    'special_rules': null,
    'start_date': '2019-05-01',
    'start_datetime': '2019-05-01T18:45:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-18T10:11:56.786364Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/europe-uefa-champions-league/2019/04/30/tottenham-vs-afc-ajax',
    'hidden': false,
    'parent': {
      'name': 'Champions League'
    },
    'id': '1943310',
    'inplay_enabled': true,
    'modified': '2019-04-18T10:11:56.786364Z',
    'name': 'Tottenham Hotspur vs. Ajax',
    'parent_id': '929817',
    'short_name': 'TOT vs. AJA',
    'slug': 'tottenham-afc-ajax-2019-04-30',
    'special_rules': null,
    'start_date': '2019-04-30',
    'start_datetime': '2019-04-30T19:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-18T10:12:03.638683Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/europe-uefa-champions-league/2019/05/01/fc-barcelona-vs-liverpool-fc',
    'hidden': false,
    'parent': {
      'name': 'Champions League'
    },
    'id': '1943311',
    'inplay_enabled': true,
    'modified': '2019-04-18T10:12:03.638683Z',
    'name': 'Barcelona vs. Liverpool',
    'parent_id': '929817',
    'short_name': 'BAR vs. LIV',
    'slug': 'fc-barcelona-liverpool-fc-2019-05-01',
    'special_rules': null,
    'start_date': '2019-05-01',
    'start_datetime': '2019-05-01T19:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-20T10:06:14.610452Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/premier-league/2019/05/04/wolverhampton-vs-fulham-fc',
    'hidden': false,
    'parent': {
      'name': 'Premier League'
    },
    'id': '1958062',
    'inplay_enabled': true,
    'modified': '2019-04-20T10:06:14.610452Z',
    'name': 'Wolverhampton Wanderers vs. Fulham',
    'parent_id': '943086',
    'short_name': 'WOL vs. FUL',
    'slug': 'wolverhampton-fulham-fc-2019-05-04',
    'special_rules': null,
    'start_date': '2019-05-04',
    'start_datetime': '2019-05-04T14:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  }
];
