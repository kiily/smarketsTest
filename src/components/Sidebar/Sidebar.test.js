import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Sidebar from './Sidebar';
import { events } from '../../mock-data/event-data';

describe('Sidebar component left side', () => {
  let component;
  let rootInstance;
  beforeEach(() => {
    component = create(<BrowserRouter><Sidebar side="left" show={true} /></BrowserRouter>);
    rootInstance = component.root;
  });

  it('should match the snapshot when rendered with props', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should render a nav if left sidebar and show is true', () => {
    const nav = rootInstance.findByType('nav');
    expect(nav.props.className).toBe('Sidebar-left in');
  });

  it('should hide the left sidebar if show is false', () => {
    component = create(<BrowserRouter><Sidebar side="left" show={false} /></BrowserRouter>);
    rootInstance = component.root;
    const nav = rootInstance.findByType('nav');
    expect(nav.props.className).toBe('Sidebar-left out');
  });
});

describe('Sidebar component right side', () => {
  let component;
  beforeEach(() => {
    component = create(<BrowserRouter><Sidebar side="right" popularEvents={events}/></BrowserRouter>);
  });

  it('should match the snapshot when rendered with props', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should render a list item tag (li) per event (5) events', () => {
    const rootInstance = component.root;
    const list = rootInstance.findByType('ul');
    expect(list.children).toHaveLength(5);
  });
});
