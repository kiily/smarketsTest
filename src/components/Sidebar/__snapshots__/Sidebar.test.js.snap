// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component left side should match the snapshot when rendered with props 1`] = `
<nav
  className="Sidebar-left in"
>
  <div
    className="sidebar-container"
  >
    <ol
      className="left-menu-quick-links"
    >
      <div
        className="links-title"
      >
        Quick Links
      </div>
      <li
        className="menu-item"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Football
        </a>
      </li>
      <li
        className="menu-item"
      >
        <a
          aria-current={null}
          href="/tennis"
          onClick={[Function]}
        >
          Tennis
        </a>
      </li>
      <li
        className="menu-item"
      >
        <a
          aria-current={null}
          href="/horse"
          onClick={[Function]}
        >
          Horse Racing
        </a>
      </li>
      <li
        className="menu-item"
      >
        <a
          aria-current={null}
          href="/politics"
          onClick={[Function]}
        >
          Politics
        </a>
      </li>
    </ol>
  </div>
</nav>
`;

exports[`Sidebar component right side should match the snapshot when rendered with props 1`] = `
<aside
  className="Sidebar-right"
>
  <div
    className="sidebar-container"
  >
    <div
      className="popular-events"
    >
      <div
        className="right-nav-header"
      >
        <span>
          Popular Events
        </span>
      </div>
      <div
        className="right-nav-content"
      >
        <ul
          className="popular-events-list"
        >
          <li>
            <a
              href="/event-detail/1925896"
              onClick={[Function]}
            >
              <span
                className="event-link-content"
              >
                <div
                  className="event-name"
                >
                  Millwall vs. Bristol City
                </div>
                <div
                  className="event-info"
                >
                  In 17 hour(s)
                </div>
              </span>
            </a>
          </li>
          <li>
            <a
              href="/event-detail/1934371"
              onClick={[Function]}
            >
              <span
                className="event-link-content"
              >
                <div
                  className="event-name"
                >
                  Swansea City vs. Derby County
                </div>
                <div
                  className="event-info"
                >
                  In 8 hour(s)
                </div>
              </span>
            </a>
          </li>
          <li>
            <a
              href="/event-detail/1943310"
              onClick={[Function]}
            >
              <span
                className="event-link-content"
              >
                <div
                  className="event-name"
                >
                  Tottenham Hotspur vs. Ajax
                </div>
                <div
                  className="event-info"
                >
                  In 17 hour(s)
                </div>
              </span>
            </a>
          </li>
          <li>
            <a
              href="/event-detail/1943311"
              onClick={[Function]}
            >
              <span
                className="event-link-content"
              >
                <div
                  className="event-name"
                >
                  Barcelona vs. Liverpool
                </div>
                <div
                  className="event-info"
                >
                  In 8 hour(s)
                </div>
              </span>
            </a>
          </li>
          <li>
            <a
              href="/event-detail/1958062"
              onClick={[Function]}
            >
              <span
                className="event-link-content"
              >
                <div
                  className="event-name"
                >
                  Wolverhampton Wanderers vs. Fulham
                </div>
                <div
                  className="event-info"
                >
                  In 3 day(s)
                </div>
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</aside>
`;
